packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -442,12 +443,28 @@ export class TokenBalancesController extends StaticIntervalPollingController<{
 
   /**
    * Override to handle our custom polling approach
+   *
+   * @param tokenSetId - The token set ID to stop polling for
    */
-  override _stopPollingByPollingTokenSetId() {
-    this.#isControllerPollingActive = false;
-    this.#requestedChainIds = []; // Clear original intent when stopping
-    this.#intervalPollingTimers.forEach((timer) => clearInterval(timer));
-    this.#intervalPollingTimers.clear();
+  override _stopPollingByPollingTokenSetId(tokenSetId: PollingTokenSetId) {
+    const parsedTokenSetId = JSON.parse(tokenSetId);
```

Comment:
Hmm okay, so we are passing in a string and parsing the contents?
Does this need to be a string?